pkgs/top-level/python-packages.nix

```diff
@@ -30195,4 +30233,162 @@ in modules // {
     };
   };
 
+  homeassistant = buildPythonApplication rec {
+    name = "${pname}-${version}";
+    pname = "homeassistant";
+    version = "0.30.2";
+
+    disabled = !isPy3k;
+
+    outputs = [ "out" ];
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/h/homeassistant/${name}.tar.gz";
+      sha256 = "1w4ngbx5wg0qjhanwcfp49gkmpmhvlqk3xy4pjv55mpzxc3l6xhf";
+    };
+
+    # I cannot get the tests to run
+    doCheck = false;
+
+    buildInputs = with self; [ wrapPython ];
+
+    propagatedBuildInputs = with self; [
+      cherrypy
+      jinja2
+      netdisco
+      pytz
+      pyyaml
+      requests2
+      sqlalchemy
+      static3
+      voluptuous
+      werkzeug
+    ];
+
+    preConfigure = ''
+      # typing is part of stdlib in python 3.5+
+      sed -i '/typing/d' setup.py
+      # allow patch level updates to voluptuous
+      sed -i -r "s/(voluptuous)==(.*)'/\1>=\2,<1'/" setup.py
+    '';
+
+    checkPhase = ''
+      PYTHONPATH=$PYTHONPATH:'.:tests' py.test
```

Comment:
typically running `py.test` is sufficient. Occasionally it helps to specify the folder with tests `py.test folder/with/tests` or `py.test $out`. Otherwise, this is fine.